demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -1368,6 +1383,9 @@ export class DemoMeetingApp implements AudioVideoObserver, DeviceChangeObserver,
 
   audioVideoDidStart(): void {
     this.log('session started');
+    if (this.enableSimulcast) {
+      this.audioVideo.addSimulcastUplinkPolicyObserver(this);
```

Comment:
The `meetingSession` holds `videoUplinkBandwidthPolicy` reference, it is overriden with the `DefaultSimulcastUplinkPolicy` once the `audioVideo's` `start()` is called, hence in order for the observers to work on the correctly assigned `SimulcastUplinkPolicy`, either I can add it here in `audioVideoDidStart` which makes sure `audioVideo` is started, else after the call: `this.audioVideo.start()`. I picked the `audioVideoDidStart` approach.

Reply:
In `initializeMeetingSession()`, the `meetingSession` holding the `videoUplinkBandwidthPolicy` does not yet have the `DefaultSimulcastUplinkPolicy` instance, the instance is created and attached in the `audioVideo.start()` call if `enabledSimulcast` is true. Hence, if I call `this.audioVideo.addSimulcastUplinkPolicyObserver(this);` in `initializeMeetingSession()` then the `videoUplinkBandwidthPolicy` is not `SimulcastUplinkPolicy` and further no access to add or remove simulcast observer methods. I can only add simulcast observer after the `audioVideo` is started.

Reply:
I understand now. Sorry, I didn't catch this scenario earlier in the former PR.

I recommend finding a way to register an observer at any point. Calling in `audioVideoDidStart()` is not good. In a builder's application, `audioVideoDidStart()` might be called more than one time if the application doesn't leave a page. The observer can be registered multiple times.

I'll think about this issue and get back to you.

Reply:
> In a builder's application, audioVideoDidStart() might be called more than one time if the application doesn't leave a page. The observer can be registered multiple times.

Actually, this is wrong since the policy will be created per start. Still, a builder shouldn't need to worry about when to register an observer.

#### Problem

The problem happens when SDK retries to connect the session. Here's an example:

1. Start a meeting.
2. In `audioVideoDidStart`, register a simulcast observer.
3. SDK retries to connect the session, e.g. due to a network problem.
  Per retry attempt, the `actionConnect` will be called. In `actionConnect`, you create a new instance of `DefaultSimulcastUplinkPolicy`. All registered observers on the previous policy will be gone.

    Note that `audioVideoDidStart` will not be invoked per retry attempt, so a builder loses a chance to add observers.

---

#### Suggestion

You can maintain an array of simulcast observers in `DefaultAudioVideoController`. In `actionConnect`, you register `DefaultAudioVideoController` as only one obsever to a newly-created `DefaultSimulcastUplinkPolicy`.

Now, you have the `encodingSimulcastLayerDidChange` method in `DefaultAudioVideoController`.

When `encodingSimulcastLayerDidChange` is called, you can call a list of simulcast obsevers.